csharp/src/Apache.Arrow/ArrowBuffer.cs

```diff
@@ -14,21 +14,37 @@
 // limitations under the License.
 
 using System;
+using System.Buffers;
 using System.Runtime.CompilerServices;
 using Apache.Arrow.Memory;
 
 namespace Apache.Arrow
 {
-    public readonly partial struct ArrowBuffer: IEquatable<ArrowBuffer>
+    public readonly partial struct ArrowBuffer : IEquatable<ArrowBuffer>, IDisposable
     {
+        private readonly IMemoryOwner<byte> _memoryOwner;
+        private readonly ReadOnlyMemory<byte> _memory;
+
         public static ArrowBuffer Empty => new ArrowBuffer(Memory<byte>.Empty);
 
         internal ArrowBuffer(ReadOnlyMemory<byte> data)
```

Comment:
Perhaps make this constructor public?

Reply:
I thought you were doing that in https://github.com/apache/arrow/pull/3971?  (BTW - I missed that you updated that PR, and approved it.)